header/header.go

```diff
@@ -47,11 +49,11 @@ func MakeExtendedHeader(
 ) (*ExtendedHeader, error) {
 	var dah DataAvailabilityHeader
 	if len(b.Txs) > 0 {
-		namespacedShares, _, err := b.Data.ComputeShares(b.OriginalSquareSize)
+		shares, err := appshares.Split(b.Data, true)
```

Comment:
In contradiction to https://github.com/celestiaorg/celestia-app/blob/dd02bb8919ec7c093784c49795c956ddb8cd2edf/pkg/shares/share_splitting.go#L24

I think this may need to be false to preserve backwards compatibility with blocks that have malleated transactions without share indexes.

```suggestion
		shares, err := appshares.Split(b.Data, false)
```

Tagging @evan-forbes for confirmation

Reply:
So we're deciding *not* to preserve block history on arabica, right?

Reply:
Yes, from our sync discussion

Reply:
Sorry I missed the sync discussion but based on comments, I think this is resolvable and we will not preserve backwards compatibility on Arabica.